addons/website_sale/static/tests/tours/website_sale_restricted_editor_ui.js

```diff
@@ -0,0 +1,42 @@
+/** @odoo-modules */
+
+import wTourUtils from 'website.tour_utils';
+
+wTourUtils.registerWebsitePreviewTour('website_sale_restricted_editor_ui', {
+    test: true,
+    url: `/shop`,
+}, [
+    {
+        content: "Open the site menu to check what is inside",
+        trigger: '[data-menu-xmlid="website.menu_site"]',
+    },
+    {
+        // Not very robust but still nice to have an extra check as this is the
+        // main purpose of this tour
+        content: "First check the user is not a designer",
+        trigger: '.dropdown-menu:has([data-menu-xmlid="website_sale.menu_product_pages"]):not(:has([data-menu-xmlid="website.menu_website_pages_list"]))',
+    },
+    {
+        content: "Ensure the publish and 'edit-in-backend' buttons are not shown",
+        trigger: '.o_menu_systray:not(:has(.o_switch_danger_success)):not(:has(.o_website_edit_in_backend))',
+        // Wait for the possibility to edit to appear
+        extra_trigger: '.o_menu_systray .o_edit_website_container a',
+    },
+    {
+        content: "Navigate to the first product",
+        trigger: 'iframe .oe_product_image_link',
+    },
+    {
+        content: "Click on publish/unpublish",
+        trigger: '.o_menu_systray .o_switch_danger_success',
+    },
+    {
+        content: "Click on edit-in-backend",
+        trigger: '.o_menu_systray .o_website_edit_in_backend a',
```

Comment:
I was going to mention that you could have waited for the publish rpc to return, and that it was trivial to do but it's broken in 16 :cry: 
Before 16, the toggle is not updated (green/red on/off) until the rpc is returned, and only according to the return, but right now it's updated before sending the RPC :cry: 

Whatever I guess, not worth if not simple (as probably useless/overthinking) :)

(I'll add smth in my todo list)

Reply:
> Before 16, the toggle is not updated (green/red on/off) until the rpc is returned, and only according to the return, but right now it's updated before sending the RPC

Note that it's a good thing... but only if the state is reverted to what it was before if the rpc fails which I guess it's not the case? :grimacing: 

> Whatever I guess, not worth if not simple (as probably useless/overthinking) :)

I don't it's useless here, it's indeed a source of race condition... I'll have a look.

Reply:
> Before 16, the toggle is not updated

Actually only the case since https://github.com/odoo/odoo/commit/2e751d80dd4b23abf374d5aa488827faaa593e62 you merged in may ^^

I am making a commit to merge the best of both worlds.